### ISSUE TITLE
regeneratorRuntime is not defined

### ISSUE THREAD

**Issue by stylixboom** [anandanand84/technicalindicators#50]
Hi, I used technicalindicators for a while, and it's quite good somehow.
However, I just upgrade to 1.0.4 today and found the issue which is not obvious to me.
Can you please check, thanks!

==============Install=============
$ npm install technicalindicators
test@1.0.0 E:\data\test
`-- technicalindicators@1.0.4
  `-- babel-polyfill@6.23.0
    +-- babel-runtime@6.23.0
    +-- core-js@2.4.1
    `-- regenerator-runtime@0.10.5

==============Run=============
$ node
var macd = require('technicalindicators').macd
ReferenceError: regeneratorRuntime is not defined
    at E:\data\test\node_modules\technicalindicators\dist\browser.js:1:16646
    at e.window (E:\data\test\node_modules\technicalindicators\dist\browser.js:1:17293)
    at Object.<anonymous> (E:\data\test\node_modules\technicalindicators\dist\browser.js:2:14270)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.require (module.js:497:17)
    at require (internal/module.js:20:19)

**Comment by anandanand84** [comment]
Added better support for nodejs. This is fixed in version 1.0.5. Thanks for pointing out.  Please verify and close. Thanks

**Comment by stylixboom** [comment]
I upgraded to 1.0.5 and it works fine right now. Seemed all compatible and no any other issue at the moment. Thank you for fixing this.

**Comment by aastein** [comment]
Happens for me on 1.0.5. @stylixboom what does your babel / webpack config look like?

**Comment by stylixboom** [comment]
@aastein webpack and babel/webpack can be another conflict, anyway, I did not use those with technicalindicators lib 1.0.5, only a plain nodejs, express, etc., as tested by the first comment.

**Comment by anandanand84** [comment]
@aastein, can you provide your webpack config or simple scenario to reproduce? I'll test.

**Comment by dandv** [comment]
@anandanand84: no webpack config necessary. Just run

    ./node_modules/.bin/webpack src/index.js dist/bundle.js

And in index.js simply have

    const SMA = require('technicalindicators').SMA;

I get the same `Uncaught ReferenceError: regeneratorRuntime is not defined` error with webpack 3.1.0.

**Comment by anandanand84** [comment]
Just to clarify this package uses babel to transpile es6 code so if you are using it in browser you should use regeneratorRuntime or use custom transpiler. If you want to use it in node or use a custom bundler like webpack we have module flag in https://github.com/anandanand84/technicalindicators/blob/master/package.json#L7 I am not familiar with webpack I use rollup. I tried your example and it did fail. A quick google search shows webpack didn't honor the module field, if we add the following in your webpack.config.js you should be good. I tested and it works fine. Let me know if it didn't work.

```
module.exports = {
  resolve: {
    mainFields: ["module", "main"]
  }
}
```

![image](https://user-images.githubusercontent.com/5538452/28025852-2061f698-655a-11e7-9bda-4403c80dac6e.png)

**Comment by aastein** [comment]
@anandanand84 confirming your solution works. You a hero.

**Comment by oeed** [comment]
I believe this mightn't yet be completely fixed. In stock standard CLI NodeJS (not in a browser or webpack) I'm still getting the same error.

I made [an example project](https://github.com/anandanand84/technicalindicators/files/1176168/Test.zip), which is simply comprised of the following two files.

_package.json_
```js
{
  "name": "example-test",
  "version": "1.0.0",
  "description": "",
  "main": "index.js",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "author": "",
  "license": "ISC",
  "dependencies": {
    "technicalindicators": "^1.0.7"
  }
}
```

_index.js_
```js
const SMA = require('technicalindicators').SMA;
```

Then running `$ npm install; node index.js` results in this error:

```
Test/node_modules/technicalindicators/dist/index.js:1
(function (exports, req
...
ReferenceError: regeneratorRuntime is not defined
```

**Comment by lteacher** [comment]
I too thought this was sorted, then suddenly I got the issue but other machine is ok. Turns out that its back in v1.0.7 whilst v1.0.5 is ok.

**Comment by stylixboom** [comment]
As people mentioned about this issue again, I think this should be re-investigated, somehow.

**Comment by anandanand84** [comment]
for now you can 

```
npm install --save babel-polyfill
```
and in just before loading techincalindicators use 

```
require('babel-polyfill')
```

We need to discuss if we need to add babel-polyfill inside library.

**Comment by lteacher** [comment]
I'm using the library in node app though so the polyfill isn't required. I'm not familiar with rollup to suggest some solution but I think since you have already nicely bundled to both `main: index.js` and `browser: browser.js` so you could for sure fix the node side of things but still discuss if you want to include the polyfill or not.

**Comment by anandanand84** [comment]
@lteacher It does require polyfill for nodejs version below 6.4. So may be I'll remove polyfill and say above 6.4

**Comment by anandanand84** [comment]
Fixed in 1.0.8 to work in nodejs without polyfill
